### PR TITLE
Improve the error message for the /share/schema/table APIs

### PR DESCRIPTION
**Pull request by kohei-tosshy**
Hello.

This is a PR for #71.

### CODE REVIEW

File: server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala [hunk 1]
```diff
@@ -62,13 +62,22 @@ class DeltaSharingServiceExceptionHandler extends ExceptionHandlerFunction {
...
-          MediaType.JSON_UTF_8,
-          JsonUtils.toJson(
-            Map(
-              "errorCode" -> ErrorCode.RESOURCE_DOES_NOT_EXIST,
-              "message" -> cause.getMessage)))
+        if (req.method().equals(HttpMethod.HEAD)) {
+          HttpResponse.of(
+            ResponseHeaders.builder(HttpStatus.NOT_FOUND)
+              .set("error-code", ErrorCode.RESOURCE_DOES_NOT_EXIST)
+              .set("message", cause.getMessage)
```
Comment (zsxwing) [inline server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:69]: perhaps not change these codes? I think it's not worth to add these non standard headers for this small change.

**Review by zsxwing — COMMENTED** [review]
Thanks for doing this! Left one minor comment.

File: server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala (hunk 1 above)
Comment (kohei-tosshy) [reply server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala]: Thank you for you comment.
I think that your opinion is correct.
I removed this two headers.

**Review by zsxwing — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**